### PR TITLE
stmtv2: ignore corrupted line when scan

### LINKED ISSUE #41373: statement v2 returns error with corrupted line
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
statements.log with line:
image

### 2. What did you expect to see? (Required)
No error returns

### 3. What did you see instead (Required)
image

### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by zhongzc**
Signed-off-by: zhongzc <zhongzc_arch@outlook.com>

### What problem does this PR solve?

Issue Number: close #41373

Problem Summary:

### What is changed and how it works?

Ignore lines that fail to unmarshal

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: util/stmtsummary/v2/reader.go [hunk 1]
```diff
@@ -388,8 +388,8 @@ func (r *HistoryReader) scheduleTasks(
 	case <-ctx.Done():
 		// notified by manager or parent ctx is canceled
 	}
-	close(rowsCh) // task done
 	mgrWg.Wait()
+	close(rowsCh) // task done
```
Comment (mornyx) [inline util/stmtsummary/v2/reader.go:392]: Is the corrupted line caused by this?

File: util/stmtsummary/v2/reader.go [hunk 2]
```diff
@@ -668,14 +671,19 @@ func (w *stmtScanWorker) putLines(
 }
 
 func (w *stmtScanWorker) readlines(reader *bufio.Reader) ([][]byte, error) {
-	firstLine, err := readLine(reader)
-	if err != nil {
-		return nil, err
-	}
+	var firstLine []byte
+	var record *stmtTinyRecord
```
Comment (mornyx) [inline util/stmtsummary/v2/reader.go:675]: Could `record` be `nil`?

**Review by mornyx — APPROVED** [review]

File: util/stmtsummary/v2/reader.go (hunk 1 above)
Comment (zhongzc) [reply util/stmtsummary/v2/reader.go:392]: No. But if err occurs, closing `rowsCh` before workers are done can cause sending to a closed channel and then panic.

**Review by crazycs520 — APPROVED** [review]

File: util/stmtsummary/v2/reader.go [hunk 3]
```diff
@@ -740,7 +748,7 @@ func (w *stmtParseWorker) run(
 func (w *stmtParseWorker) handleLines(
 	lines [][]byte,
 	rowsCh chan<- [][]types.Datum,
-	errCh chan<- error,
+	_ chan<- error,
```
Comment (tangenta) [inline util/stmtsummary/v2/reader.go:751]: Why not remove it?

Comment (zhongzc) [reply util/stmtsummary/v2/reader.go:751]: In case it becomes needed in future

**Pull request merged**